### PR TITLE
Fix allow allow_non_public_ip & add tests

### LINKED ISSUE #400: Errors in fileInfo and initialize endpoints in Ropsten network
**Describe the bug**
It's not possible to buy or publish an asset on Robsten right now due errors in `/fileInfo` and `/initialize`

Edit: 
The errors are happening in `allow_non_public_ip()` when "allow_non_public_ip" is a string like "False"

```
2022-03-17 17:02:46 provider-7d5cdd5999-lgmcw ocean_provider.run[14] INFO incoming request = http, GET, 10.18.67.111, /?
ValueError: invalid literal for int() with base 10: 'False'
    return bool(int(self.get("resources", NAME_ALLOW_NON_PUBLIC_IP, fallback=0)))
  File "/ocean-provider/ocean_provider/config.py", line 138, in allow_non_public_ip
    allow_non_public_ip = get_config().allow_non_public_ip
...
```

**To Reproduce**
We can use this existing asset: https://market-git-v4-oceanprotocol.vercel.app/asset/did:op:37349027ea77afd391e9f39471582d9093a8eb22969d8f320cdba016f84219fa an try to see the datatoken price (it doesn't work). Open the console tab, you will find errors in `/fileInfo` and `/initialize` enpoints.

Alternatively try to publish an asset in Ropsten.

These are the failing http requests for the existing asset `did:op:37349027ea77afd391e9f39471582d9093a8eb22969d8f320cdba016f84219fa:`

Run
```
curl 'https://v4.provider.ropsten.oceanprotocol.com/api/services/initialize?documentId=did:op:37349027ea77afd391e9f39471582d9093a8eb22969d8f320cdba016f84219fa&serviceId=670015bec0c85c25a1b29c1d7960916ded9a8d1a9c2c4b1bc5a4b7db8ddc1f68&fileIndex=0&consumerAddress=0x0000000000000000000000000000000000000000' \
  -H 'authority: v4.provider.ropsten.oceanprotocol.com' \
  -H 'sec-ch-ua: " Not A;Brand";v="99", "Chromium";v="99", "Google Chrome";v="99"' \
  -H 'sec-ch-ua-mobile: ?1' \
  -H 'user-agent: Mozilla/5.0 (Linux; Android 6.0; Nexus 5 Build/MRA58N) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Mobile Safari/537.36' \
  -H 'sec-ch-ua-platform: "Android"' \
  -H 'content-type: application/json' \
  -H 'accept: */*' \
  -H 'origin: https://market-git-v4-oceanprotocol.vercel.app' \
  -H 'sec-fetch-site: cross-site' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-dest: empty' \
  -H 'referer: https://market-git-v4-oceanprotocol.vercel.app/' \
  -H 'accept-language: en,es-ES;q=0.9,es;q=0.8,fr;q=0.7' \
  --compressed
```
Result:
```json
{"context":{"consumerAddress":"0x0000000000000000000000000000000000000000","documentId":"did:op:37349027ea77afd391e9f39471582d9093a8eb22969d8f320cdba016f84219fa","fileIndex":"0","serviceId":"670015bec0c85c25a1b29c1d7960916ded9a8d1a9c2c4b1bc5a4b7db8ddc1f68"},"error":"invalid literal for int() with base 10: 'False'"}
```
Run:
```
curl 'https://v4.provider.ropsten.oceanprotocol.com/api/services/fileinfo' \
  -H 'authority: v4.provider.ropsten.oceanprotocol.com' \
  -H 'sec-ch-ua: " Not A;Brand";v="99", "Chromium";v="99", "Google Chrome";v="99"' \
  -H 'sec-ch-ua-mobile: ?1' \
  -H 'user-agent: Mozilla/5.0 (Linux; Android 6.0; Nexus 5 Build/MRA58N) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Mobile Safari/537.36' \
  -H 'sec-ch-ua-platform: "Android"' \
  -H 'content-type: application/json' \
  -H 'accept: */*' \
  -H 'origin: https://market-git-v4-oceanprotocol.vercel.app' \
  -H 'sec-fetch-site: cross-site' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-dest: empty' \
  -H 'referer: https://market-git-v4-oceanprotocol.vercel.app/' \
  -H 'accept-language: en,es-ES;q=0.9,es;q=0.8,fr;q=0.7' \
  --data-raw '{"did":"did:op:37349027ea77afd391e9f39471582d9093a8eb22969d8f320cdba016f84219fa","serviceId":"670015bec0c85c25a1b29c1d7960916ded9a8d1a9c2c4b1bc5a4b7db8ddc1f68"}' \
  --compressed
  ```
  Result:
  ```json
{"context":{"did":"did:op:37349027ea77afd391e9f39471582d9093a8eb22969d8f320cdba016f84219fa","serviceId":"670015bec0c85c25a1b29c1d7960916ded9a8d1a9c2c4b1bc5a4b7db8ddc1f68"},"error":"invalid literal for int() with base 10: 'False'"}
  ```

Desktop
 - OS: Windows 10
 - Browser: Firefox
 - Version 98.0 64 bit

### PR DESCRIPTION
**Pull request by md0x**
Signed-off-by: Pablo Maldonado <pablo@oceanprotocol.com>

Fixes #400  .

Changes proposed in this PR:

-  Fix errors in `allow_non_public_ip` from `config.py`

### CODE REVIEW

**Review by alexcos20 — APPROVED** [review]

File: ocean_provider/test/test_config.py [hunk 1]
```diff
@@ -50,3 +50,66 @@ def test_config_dict():
...
+    config = Config(options_dict=config_dict)
+    assert config.allow_non_public_ip == True
+
+    config_dict = {
+        "eth-network": {},
+        "resources": {
+            "aquarius.url": "https://another-aqua2.url",
+            "allow_non_public_ip": 1,
+        },
+    }
```
Comment (DMats) [inline ocean_provider/test/test_config.py:113]: Please add a test case where `allow_non_public_ip` is not set (and thus would fallback to `None`).  

From a design standpoint - what should happen if `allow_non_public_ip` is not set?  I'm thinking when `allow_non_public_ip` is not set, then non-public IPs should not be allowed.

**Review by DMats — CHANGES_REQUESTED** [review]

File: ocean_provider/test/test_config.py [hunk 2]
```diff
@@ -50,3 +50,66 @@ def test_config_dict():
     }
     config = Config(options_dict=config_dict)
     assert config.aquarius_url == "https://another-aqua2.url"
+
+
+@pytest.mark.unit
+def test_allow_non_public_ip():
+    config_dict = {
```
Comment (calina-c) [inline ocean_provider/test/test_config.py:57]: Looks like we could simplify these tests by only modifying the allow... var în the next iterations. I.e. Instead of redefining config dict, we on y assign that key

**Review by calina-c — CHANGES_REQUESTED** [review]
I've also sent some useful stuff in Slack 😁

**Review by alexcos20 — APPROVED** [review]

**Review by alexcos20 — APPROVED** [review]

File: ocean_provider/test/test_config.py (hunk 1 above)
Comment (alexcos20) [reply ocean_provider/test/test_config.py]: this should be enough:  https://github.com/oceanprotocol/provider/blob/v4main/Dockerfile#L52
